### PR TITLE
[ocm-clusters] add support for autoscaling and cluster update

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-2917

depends on https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/14339

### CODE REVIEW

File: reconcile/ocm_clusters.py [hunk 1]
```diff
@@ -12,12 +12,16 @@
...
 
-def fetch_current_state(clusters):
+def fetch_desired_state(clusters):
     desired_state = {c['name']: {'spec': c['spec'], 'network': c['network']}
                      for c in clusters}
     # remove unused keys
     for desired_spec in desired_state.values():
         desired_spec['spec'].pop('upgrade', None)
+        # remove empty keys in spec
+        empty_keys = [k for k, v in desired_spec['spec'].items() if v is None]
```
Comment (apahim) [inline reconcile/ocm_clusters.py:22]: you can use a dict comprehension and avoid the pop ahead... `{k: n for k, v in d.items() if v is not None }`

File: reconcile/utils/ocm.py [hunk 2]
```diff
@@ -140,12 +147,60 @@ def create_cluster(self, name, cluster, dry_run):
...
+        api = f'/api/clusters_mgmt/v1/clusters/{cluster_id}'
+        cluster_spec = cluster['spec']
+        ocm_spec = {
+            'nodes': {
+                'compute_machine_type': {
+                    'id': cluster_spec['instance_type']
+                }
+            },
+            'storage_quota': {
+                'value': float(cluster_spec['storage'] * pow(1024, 3))
```
Comment (apahim) [inline reconcile/utils/ocm.py:182]: no need to calc `1024**3` all the time... let's use the actual number and add a commet... `... * 1073741824  # 1024 ** 3`

**Review by apahim — APPROVED** [review]
small nitpics that can safelly be ignored. lgtm, thanks!

**Comment by apahim** [comment]
Sorry, for the first time I clicked "Update branch" and it did the worst thing it could do :(
Running `git push -f` should fix it

File: reconcile/ocm_clusters.py (hunk 1 above)
Comment (maorfr) [reply reconcile/ocm_clusters.py:22]: absolutely!

File: reconcile/utils/ocm.py (hunk 2 above)
Comment (maorfr) [reply reconcile/utils/ocm.py:182]: makes sense

**Pull request merged**